### ISSUE TITLE
Error when trying to do quest stuff on a server.

### ISSUE THREAD

**Issue by thedeerirty** [Rebirth-of-the-Night/ToroQuestRevamped#14]
![image](https://user-images.githubusercontent.com/58305182/133793818-9aa0b345-ff94-4195-89f8-398ab8dfbe51.png)

Player's can't complete or abandon quests, and this is what the server flashes with when trying.

**Comment by bglandolt** [comment]
Which quest showed up when this happened? (build quest was bugged in earlier versions)
Which version of the mod are you using?
What is your rep?

**Comment by thedeerirty** [comment]
I'm not sure, this wasn't me doing this, I just noted it in the code.
This seems to be an issue on my entire server though, not just with your mod.
![image](https://user-images.githubusercontent.com/58305182/134088186-cbbe8d49-f4b6-472d-83b4-1811eebb3f40.png)
here's an example of the same sort of error appearing 
I have noticed guard chat messages don't appear on the server too. It's odd.

**Comment by thedeerirty** [comment]
[latest.log](https://github.com/Rebirth-of-the-Night/ToroQuestRevamped/files/7205203/latest.log)
here's a log that has a lot of issues I'm seeing

**Comment by bglandolt** [comment]
Yeah it seems like you have a lot of issues (not just with toroquest)

"I have noticed guard chat messages don't appear on the server too" > Yeaaah I can see where I messed up, I was using a method that was client-side only. I apologize for the inconvenience, working on a fix rn

**Comment by thedeerirty** [comment]
Would you happen to be able to identify a sort of root cause of the issues as a whole, besides just your mod? Or is it your mod in relation to other mods that's causing a bit of issue?

It's mostly weird cause it's solely a serverside issue; I've tested and can run all of this perfectly fine in just a singleplayer world

**Comment by bglandolt** [comment]
https://mcforge.readthedocs.io/en/latest/concepts/sides/

"This mistake can also be made explicitly by accessing physical client-only classes such as Minecraft from common code that runs or can run on the logical server. **This mistake is easy to miss for beginners who debug in a physical client. The code will work there, but it will immediately crash on a physical server**."

Most debugging is done in the physical client so it is easy for modders (such as myself) to sometimes mess up/ miss something

**Comment by thedeerirty** [comment]
Yeah I understand; I too work on minecraft mods, though I'm the artist of my projects and not the coder, so I just know a bit here n' there.

Would your mod using a method like this affect other mods in the server who are having similar issues? I notice that yours, and Beyond the Veil were both throwing this error, with more or less the same sort of wording/reporting so I'm curious if its just one or both of them

**Comment by bglandolt** [comment]
I am fairly certain my mod wouldn't affect another one, it's most likely the owner of beyond the veil (and other mods that are giving you errors) hasnt fully tested his server side (like me). It is kinda obnoxious to test haha I wonder if there is a better/ faster way. Most of us arent professionals so our mods are bound to have bugs, especially the bigger ones. I would just let the owner of BtV know

**Comment by thedeerirty** [comment]
I'll inform em, yeah
I'm not sure how the Dev I work with does it, nor if there'd be a different method for 1.12 vs 1.15/1.16 server testing

**Comment by thedeerirty** [comment]
yo! did the recent update patch things up okay?

**Comment by bglandolt** [comment]
I was testing it for about an hour and everything seemed stable, a few things dont work like bandits cant use boats on a server but im working on that rn
